src/common/components/ConfirmMenuItem.js

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright © 2021-2023 Technology Matters
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published
+ * by the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program. If not, see https://www.gnu.org/licenses/.
+ */
+import React, { useCallback, useEffect, useState } from 'react';
+import { MenuItem } from '@mui/material';
+
+import ConfirmationDialog from 'common/components/ConfirmationDialog';
+
+const ConfirmMenuItem = props => {
+  const [openConfirmation, setOpenConfirmation] = useState(false);
+  const { confirmTitle, confirmMessage, confirmButton, onConfirm, children } =
+    props;
+
+  useEffect(() => {
+    setOpenConfirmation(false);
+  }, []);
+
+  const onClick = useCallback(event => {
+    setOpenConfirmation(true);
+    event.stopPropagation();
```

Comment:
@david-code yes, I think that is common, there are some cases where you have the same listener on a parent component and you want to avoid the event to keep propagating